### DISCUSSION TITLE
.gitignore separado por remote

### DISCUSSION THREAD

**Discussion by adeonir** [frontendbr/forum#236] [Perguntas e Respostas]
Ae pessoal, tudo certo?

Tenho um git remote de produção e um remote no bitbucket.
Minha dúvida é se tem como separar usando gitignore. No bitbucket vou subir todos os src e na produção somente o dist.

**Comment by herberthenrique** [comment]
Cara, a partir do ponto que os arquivos estão no repositório e foram commitados, o gitignore não irá retira-los do repositório mais.
O que você pode fazer é uma outra estrategia, sendo a mais elegante usando CI e um task runner como gulp ou grunt.
No CI, assim que iniciar o processo de build você usa o task runner para criar o `dist/` 
E então o pacote que o CI irá gerar para o servidor de produção será somente a pasta `dist`  (:

Recomenda dar uma olhada no [Travis](https://travis-ci.org), meu favorito!
Mas existem diversos, como o jenkins, bamboo, teamcity, VSO, etc...

**Comment by yakseok** [comment]
Tu pode também usar git hook. Toda vez que o repositório "produção" que tu mencionou receber um commit/push, ele executar o hook. Nesse hook tu pode iniciar o processo de build, seja usando um software pra isso ou um simples script, vai depender do teu problema.

Uma vez eu precisei fazer isso e pesquisei exatamente a mesma coisa: como usar um gitignore para "remotes" diferentes xD e acabei achando essa solução. Na época usei git hook + phing.

Só deixando essa ideia também, pode ser útil pra alguém! :smile:
